base/common/wide_integer_impl.h

```diff
@@ -275,23 +283,19 @@ struct wide_integer<Bits, Signed>::_impl
         return lhs;
     }
 
-    constexpr static wide_integer<Bits, signed> shift_left(const wide_integer<Bits, signed> & rhs, int n)
+    constexpr static integer<Bits, signed> shift_left(const integer<Bits, signed> & rhs, int n) noexcept
     {
-        // static_assert(is_negative(rhs), "shift left for negative lhsbers is underfined!");
-        if (is_negative(rhs))
-            throw std::runtime_error("shift left for negative lhsbers is underfined!");
```

Comment:
Crach was here when called from `operator << () noexcept`

Reply:
By the way, what is the point of `noexcept` there? I always thought its only function was making the code slower by inserting try/catch + std::terminate()...

Reply:
On an arithmetic operator especially, it must be costing us a lot.

Reply:
I remove it in this PR and make `shift_left`noexcept.

Reply:
Current library logic is to have `noexcept` all the places its possible. Probably we should change it for better performance but not in this PR.